pkg/util/sha.go

```diff
@@ -33,9 +34,46 @@ func PrefixSHA(sha string) string {
 		prefix = "sha1:"
 	case 64:
 		prefix = "sha256:"
+	case 96:
+		prefix = "sha384:"
 	case 128:
 		prefix = "sha512:"
 	}
 
 	return fmt.Sprintf("%v%v", prefix, sha)
 }
+
+func UnprefixSHA(sha string) (crypto.Hash, string) {
+	components := strings.Split(sha, ":")
+
+	if len(components) == 2 {
+		prefix := components[0]
+		sha = components[1]
+
+		switch prefix {
+		case "sha1":
+			return crypto.SHA1, sha
+		case "sha256":
+			return crypto.SHA256, sha
+		case "sha384":
+			return crypto.SHA384, sha
+		case "sha512":
+			return crypto.SHA512, sha
+		default:
+			return crypto.Hash(0), ""
+		}
+	}
+
+	switch len(sha) {
+	case 40:
+		return crypto.SHA1, sha
```

Comment:
I know this is probably because the function above included `sha1`, but I think we should remove this (and the stanza above). There's no other use of that AFAIK in sigstore, and I wouldn't want to perpetuate this.

Reply:
actually, i think it is used to search for entries from gitsign, im working to confirm 

@wlynch - any thoughts?

Reply:
https://github.com/sigstore/rekor/pull/499

Reply:
nm, i think we need to leave this - but it would be nice to add a test case that calls `CreateFromArtifactProperties` with a proposed sha1 value to ensure that it doesn't somehow end up in the log.

Reply:
Added it in https://github.com/sigstore/rekor/pull/1959/commits/6903f5f774fb0fed31211bee18193d17cf6c11ab . Please let me know if this is what you meant, otherwise I can adjust it.

Reply:
We don't use this for gitsign today - gitsign does a sha256 over the commit content + sha1 for rekor.

(though it would be a neat way to remove some of the chattiness gitsign has today with it's sha -> rekor entry lookup 🤔 )